fe/fe-core/src/main/java/org/apache/doris/nereids/analyzer/UnboundOneRowRelation.java

```diff
@@ -0,0 +1,121 @@
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.analyzer;
+
+import org.apache.doris.nereids.exceptions.UnboundException;
+import org.apache.doris.nereids.memo.GroupExpression;
+import org.apache.doris.nereids.properties.LogicalProperties;
+import org.apache.doris.nereids.properties.UnboundLogicalProperties;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.NamedExpression;
+import org.apache.doris.nereids.trees.expressions.Slot;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.PlanType;
+import org.apache.doris.nereids.trees.plans.algebra.OneRowRelation;
+import org.apache.doris.nereids.trees.plans.logical.LogicalLeaf;
+import org.apache.doris.nereids.trees.plans.visitor.PlanVisitor;
+import org.apache.doris.nereids.util.Utils;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * A relation that contains only one row consist of some constant expressions.
+ * e.g. select 100, 'value'
+ */
+public class UnboundOneRowRelation extends LogicalLeaf implements Unbound, OneRowRelation {
+    private final List<NamedExpression> projects;
+
+    public UnboundOneRowRelation(List<NamedExpression> projects) {
+        this(projects, Optional.empty(), Optional.empty());
+    }
+
+    private UnboundOneRowRelation(List<NamedExpression> projects, Optional<GroupExpression> groupExpression,
+            Optional<LogicalProperties> logicalProperties) {
+        super(PlanType.LOGICAL_UNBOUND_ONE_ROW_RELATION, groupExpression, logicalProperties);
+        Preconditions.checkArgument(projects.stream().allMatch(Expression::isConstant),
+                "OneRowRelation must consist of some constant expression");
+        this.projects = ImmutableList.copyOf(projects);
+    }
+
+    @Override
+    public <R, C> R accept(PlanVisitor<R, C> visitor, C context) {
+        return visitor.visitUnboundOneRowRelation(this, context);
+    }
+
+    @Override
+    public List<NamedExpression> getProjects() {
+        return projects;
+    }
+
+    @Override
+    public List<Expression> getExpressions() {
+        throw new UnsupportedOperationException(this.getClass().getSimpleName() + " don't support getExpression()");
```

Comment:
why could not return projects?

Reply:
Keep same as the logic: UnboundXxx can not invoke getExpressions(), like UnboundRelation.
It maybe help up to find some bug if we getExpression from an unbound plan.

Reply:
just curiousness, i think current code is ok